bitmark-references/bitmark-node-software/bitmark-blockain.md

```diff
@@ -0,0 +1,125 @@
+# Blockchain
+
+## Block
+
+Block is a bunch of transactions that have been added to blockchain. A bitmark block contains data as follows:
+
+  size (bytes)    data
+ --------------  ----------------------------------------------------------------
+   2              block version number
+   2              transaction count inside the block
+   8              block number (block height)
+  32              argon2 hash of previous block header
+  32              sha3 hash of all transactions in the block (merkle root)
+   8              timestamp of block generation time, since 1970-01-01T00:00 UTC
+   8              current difficulty in compact mode
+   8              number to make hash meets difficulty
+   ... to end     transactions
+
+Different types of transaction include different length and content inside.
+
```

Comment:
Examples are always wonderful. It'd be great to show a sample blocker header of an existing block and highlight all the pieces.

Reply:
It's quite complicated, but I have list an example to demonstrate fiellds

Reply:
Yep, complicated, but it'll be great for someone really trying to figure this out.

I'd suggest also showing it broken up:

For example, consider the block header string of:

"05008300a1820000000000001e886357e769dab9c5f9b68a9101c13888a8b9aaf2bc21ff3aa2b78fbac114002684b245a1f51f54677770530032471f0ec4c36ffae5f14001eb15f42082e31becc1895d00000000d3dbb256c7042f00deb653d83b4a0bc7"

Remember that a byte is made up of two hexadecimal characters. The header string could thus be broken up as follows, using the byte counts above:

"0500/8300/a182000000000000/1e886357e769dab9c5f9b68a9101c13888a8b9aaf2bc21ff3aa2b78fbac11400/2684b245a1f51f54677770530032471f0ec4c36ffae5f14001eb15f42082e31b/ecc1895d00000000/d3dbb256c7042f00/deb653d83b4a0bc7"

Also remember that each element has its bytes reversed because it's encoded in little endian. The various fields are thus decoded as follows:

Version:0x5,
TransactionCount:0x83,
Number:0x82a1,
PreviousBlock: 0x0014c1ba8fb7a23aff21bcf2aab9a88838c101918ab6f9c5b9da69e75763881e,
MerkleRoot: 0x1be38220f415eb0140f1e5fa6fc3c40e1f47320053707767541ff5a145b28426,
Timestamp:0x5d89c1ec,
Difficulty:00978263ab596de9800000000000000000000000000000000000000000000000,
Nonce:0xc70b4a3bd853b6de

Reply:
changed, thanks for wording